go.mod

```diff
@@ -4,10 +4,11 @@ go 1.20
 
 require (
 	github.com/golang/mock v1.6.0
+	github.com/google/go-cmp v0.6.0
```

Comment:
Is this new dependency really required for everyone using this library? Nobody downstream will be able to update to 0.7.0 when it comes out before we do it because we've forced this on them (Go mod rules say minor of 0-ver not compatible w/ next minor of 0-ver).

Reply:
This is used by the official protojson code for the most part, so our downstreams are already stuck with it. I simply added it here so the test was better:
```
$ rg -t go -l go-cmp
internal/protojson/json/decode_test.go
internal/protojson/json/encode_test.go
internal/protojson/order/order_test.go
internal/temporalcommonv1/payload_json_test.go
temporalproto/json_marshal_test.go
temporalproto/compatibility_test.go
```

Reply:
Our downstreams can update Go proto independently if/when Go proto updates and also updates this library. But us now taking an explicit dependency on it means they can't, we have to update this dependency. I'd really like to keep this a transitive dependency instead of an explicit one. Do we have to use it?

Reply:
If we rewrite all the protojson tests that use it then no. I _could_ just delete those tests as well, but I'd like to be sure this continues working.

I'll see how much pain rewriting those checks would be

Reply:
Thanks! I know it seems a bit overly cautious, but definitely feels safer to not infect our downstream users with our test-only dependency

Reply:
I rewrote them. You'll see the effect in the other PR